### DISCUSSION TITLE
Impressive project ! Here are 2 ideas

### DISCUSSION THREAD

**Discussion by beber666** [impredicative/podgenai#1] [Ideas]
Thank you this project is really great.

I've been able to generate a great podcast in french !

Here are 2 ideas:
- the ability to add a parameter to load a script text file, instead of automaticaly generate it by OpenAI
- The ability to limit to 10 minutes by example the text genrated by OpenAI
- The ability to remove "Section 1, Section 2 etc...) in the audio podcast

Good luck !

**Comment by impredicative** [comment]
@beber666 Thanks for the feedback. I am glad to hear that you were able to generate a file in a language other than English.

Regarding loading a text file, I am sorry but that is definitely outside the scope of this project. I will think about adding a script for it, but it's very low priority. Meanwhile, you can always manually edit the output in the `work` directory before generating the final audio output. podgenai will then use the edited text. If you want a section to be regenerated, you can delete the corresponding section file from the `work` directory, then rerun the podgenai command.

Regarding limiting to `n` minutes, this is impractical to do properly, but it can be more practical for me to add an option to limit the maximum number of output sections to say 5. I have implemented a feature like this called `--max-sections` in my other project [newssurvey](https://github.com/impredicative/newssurvey). This could produce a shorter file if that's the goal. I will certainly consider this modified request.

Regarding the ability to remove the section markers, this is now implemented in version 0.5.3 using the `--no-markers` option. This will also move the English disclaimer to the end of the file. I tried generating two foreign language files with this option, and they came out okay. Please refer to the word "foreign" wherever it is present in the readme. For French language generation, here is a sample output for you for "[L'histoire de Napoléon Bonaparte (français)](https://mega.nz/file/FB9W3KhD#ReEnFmKA90EK9_OwcUNBkMa47H3ptgG4yOKPFQW2iKM)". As you see, this topic itself is in French, and there is also the " (français)" suffix which is practically necessary. If a foreign language topic is rejected, just it try again.

**Comment by impredicative** [comment]
To update, a `--max-sections` option has now been added. Its value can be between 3 and 100. If it's set to 3, the generated audio should be close to 10 minutes long. I discourage its use, however.

**Reply by beber666** [reply]
Thank you for the update !

**Comment by impredicative** [comment]
By the way, podgenai has now been updated to use the newer GPT model gpt-4o-2024-08-06. This had to be done because the older model lacked certain recent information that only the newer model has.

A side effect is that all episodes generated with this newer model are automatically smaller.
